apiserver/facades/controller/cleaner/register.go

```diff
@@ -24,12 +24,15 @@ func newCleanerAPI(ctx facade.ModelContext) (*CleanerAPI, error) {
 	if !authorizer.AuthController() {
 		return nil, apiservererrors.ErrPerm
 	}
+
+	serviceFactory := ctx.ServiceFactory()
 	return &CleanerAPI{
 		st:             getState(ctx.State()),
 		resources:      ctx.Resources(),
 		objectStore:    ctx.ObjectStore(),
-		machineRemover: ctx.ServiceFactory().Machine(),
-		appRemover:     ctx.ServiceFactory().Application(),
-		unitRemover:    ctx.ServiceFactory().Unit(),
+		machineRemover: serviceFactory.Machine(),
+		// For removing applications, we don't need a storage registry.
+		appRemover:  serviceFactory.Application(nil),
```

Comment:
If the registry is optional, is it better to cache the registry in the facade and then pass it in via service method arg on demand?

Reply:
The registry depends on the environ for the model. It's created up front during the register process and handed off to the services that need it, which are also constructed during register. If nil is passed in, then a "Not Implemented" registry is used which will error for any calls made to it. Many facade methods will require the registry so it will bloat all their signatures as opposed to setting it up when the service is constructed. For this cleaner facade, the only service method needed is to remove an application. We could create the registry and pass it in, but that's a lot of overhead.

Reply:
ok.